src/awkward/_v2/contents/unionarray.py

```diff
@@ -701,7 +701,13 @@ def _argsort_next(
         kind,
         order,
     ):
+        if len(self) == 0:
+            return ak._v2.contents.NumpyArray(self.nplike.empty(0, np.int64))
+
         simplified = self.simplify_uniontype(mergebool=True)
+        if len(simplified) == 0:
+            return ak._v2.contents.NumpyArray(self.nplike.empty(0, np.int64))
```

Comment:
(This, and a similar statement above.) A value, such as the length being zero, should not determine the type of an array, such as "int". This would rightly fail a type-tracer test, which might not know that the length is zero. Simplify turns `union[X, X]` into `X` and `union[union[X, Y], Z]` into `union[X, Y, Z]`, but only things if those types, determined by type.

Reply:
Sorry—I didn't realize this was in argaort. The above code is fine if they're returning the expected result of the argsort: an empty int-type array. If that's the case you can resolve this.

Reply:
Thanks! Yes I enabled the union array tests and saw that it needs fixing.